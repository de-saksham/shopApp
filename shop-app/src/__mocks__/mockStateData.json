{
    "reducer": {
        "ProductReducer": {
            "isLoading": false,
            "isSuccess": true,
            "totalPages": 5,
            "currentPage": 1,
            "activeCategory": "FRUITS",
            "isCartEmpty": false,
            "isModalActive": false,
            "orderStatus": true,
            "products": [
                {
                    "_id": "642899de1d0bfcc11aa6778a",
                    "title": "Washington Apple",
                    "desc": "Juicy red apples straight from United States of America",
                    "price": 7.99,
                    "stock": 2,
                    "category": "FRUITS"
                },
                {
                    "_id": "642899de1d0bfcc11aa6778b",
                    "title": "Green Apple",
                    "desc": "The Granny Smith, also known as a green apple or sour apple",
                    "price": 19.99,
                    "stock": 124,
                    "category": "FRUITS"
                }
            ],
            "cart": [
                {
                    "_id": "642899de1d0bfcc11aa6778a",
                    "title": "Washington Apple",
                    "desc": "Juicy red apples straight from United States of America",
                    "price": 7.99,
                    "stock": 2,
                    "category": "FRUITS",
                    "quantity": 1
                },
                {
                    "_id": "642899de1d0bfcc11aa6778b",
                    "title": "Green Apple",
                    "desc": "The Granny Smith, also known as a green apple or sour apple",
                    "price": 19.99,
                    "stock": 124,
                    "category": "FRUITS",
                    "quantity": 11
                }
            ]
        },
        "productApi": {
            "data": [
                {
                    "_id": "642899de1d0bfcc11aa6778a",
                    "title": "Washington Apple",
                    "desc": "Juicy red apples straight from United States of America",
                    "price": 7.99,
                    "stock": 2,
                    "category": "FRUITS"
                },
                {
                    "_id": "642899de1d0bfcc11aa6778b",
                    "title": "Green Apple",
                    "desc": "The Granny Smith, also known as a green apple or sour apple",
                    "price": 19.99,
                    "stock": 124,
                    "category": "FRUITS"
                }
            ]
        },
        "orderApi": {
            "data": {
                "order": [{
                    "itemId": "642899de1d0bfcc11aa6778b",
                    "quantity": 2
                }],
                "userId": 1
            }
        }
    }
}