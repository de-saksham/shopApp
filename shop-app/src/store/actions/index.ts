export {
    init,
    isLoading, 
    storeCategory,
    storeProducts,
    addToCart,
    updateCurrentPage,
    isModalActive,
    orderStatus
} from './products';