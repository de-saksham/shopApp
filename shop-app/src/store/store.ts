import { configureStore } from "@reduxjs/toolkit";
import { combineReducers } from 'redux'
import { productApi } from "../controllers/productController";
import { orderApi } from "../controllers/orderController";
import ProductReducer, { initialState } from './reducers/products';

const reducer = combineReducers({
    ProductReducer,
});

export const store = configureStore({
    reducer: {
        [productApi.reducerPath]: productApi.reducer,
        [orderApi.reducerPath]: orderApi.reducer,
        reducer
    },
    middleware: (getDefaultMiddleware: any) => getDefaultMiddleware().concat([productApi.middleware, orderApi.middleware])
});
