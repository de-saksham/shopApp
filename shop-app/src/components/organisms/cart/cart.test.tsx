import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { configure, mount } from 'enzyme';
import * as ShallowRenderer from 'react-test-renderer/shallow';
import { store } from '../../../store/store';
import mockStateData from '../../../__mocks__/mockStateData.json';
import Cart from './cart';
import { Provider } from 'react-redux';

configure({ adapter: new Adapter() });


describe('Cart - Test Suite', () => {
    let shallowRenderer: any;

    beforeEach(() => {
        shallowRenderer = ShallowRenderer.createRenderer();
    });

    test('Snapshot for Cart', () => {
        shallowRenderer.render(
            <Provider store={store}>
                <Cart />
            </Provider>
        );
        expect(shallowRenderer.getRenderOutput()).toMatchSnapshot();
    });
});

describe('Cart', () => {
    const wrapper = mount(
        <Provider store={store}>
            <Cart />
        </Provider>
    );

    it('Main Cart should be rendered', () => {
        const title = wrapper.find('mainCart');
        expect(title).toBeTruthy();
    });

    it('CartItems should be rendered', () => {
        const CartItems = wrapper.find('CartItems');
        expect(CartItems).toHaveLength(1);
    });
});