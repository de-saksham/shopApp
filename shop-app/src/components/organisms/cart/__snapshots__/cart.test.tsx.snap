// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart - Test Suite Snapshot for Cart 1`] = `
<Context.Provider
  value={
    Object {
      "getServerState": undefined,
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <Cart />
</Context.Provider>
`;
